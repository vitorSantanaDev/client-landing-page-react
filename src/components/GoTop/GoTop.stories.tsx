import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import GoTop from '.'

export default {
  title: 'GoTop',
  component: GoTop,
  args: {}
} as ComponentMeta<typeof GoTop>

export const Default: ComponentStory<typeof GoTop> = (args) => (
  <div style={{ height: '300vh' }}>
    <h1>Lorem</h1>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Autem, quidem
      ullam perferendis assumenda magni blanditiis recusandae omnis nostrum
      doloremque voluptatibus illo culpa quisquam nisi. Cupiditate quam sint rem
      voluptate eveniet!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Autem, quidem
      ullam perferendis assumenda magni blanditiis recusandae omnis nostrum
      doloremque voluptatibus illo culpa quisquam nisi. Cupiditate quam sint rem
      voluptate eveniet!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Autem, quidem
      ullam perferendis assumenda magni blanditiis recusandae omnis nostrum
      doloremque voluptatibus illo culpa quisquam nisi. Cupiditate quam sint rem
      voluptate eveniet!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Autem, quidem
      ullam perferendis assumenda magni blanditiis recusandae omnis nostrum
      doloremque voluptatibus illo culpa quisquam nisi. Cupiditate quam sint rem
      voluptate eveniet!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Autem, quidem
      ullam perferendis assumenda magni blanditiis recusandae omnis nostrum
      doloremque voluptatibus illo culpa quisquam nisi. Cupiditate quam sint rem
      voluptate eveniet!
    </p>
    <GoTop {...args} />
  </div>
)
