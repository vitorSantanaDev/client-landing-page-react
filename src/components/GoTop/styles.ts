import styled, { css } from 'styled-components'

export const Wrapper = styled.a`
  ${({ theme }) => css`
    width: 4rem;
    height: 4rem;
    border-radius: 50%;

    display: flex;
    align-items: center;
    justify-content: center;

    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;

    color: ${theme.colors.white};
    background-color: ${theme.colors.primaryColor};
  `}
`
