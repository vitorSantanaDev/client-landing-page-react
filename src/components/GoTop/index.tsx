import React from 'react'

import { KeyboardArrowUp } from '@styled-icons/material-outlined/KeyboardArrowUp'
import * as S from './styles'

const GoTop: React.FC = () => (
  <S.Wrapper aria-label="Go to top" href="#" title="Go to top">
    <KeyboardArrowUp />
  </S.Wrapper>
)

export default GoTop
