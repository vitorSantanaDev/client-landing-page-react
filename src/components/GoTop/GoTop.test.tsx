import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import GoTop from '.'

describe('<GoTop />', () => {
  it('should render a go to top', () => {
    renderTheme(<GoTop />)
    const link = screen.getByRole('link', { name: /go to top/i })
    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('href', '#')
  })

  it('should match snapshot', () => {
    const { container } = renderTheme(<GoTop />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
