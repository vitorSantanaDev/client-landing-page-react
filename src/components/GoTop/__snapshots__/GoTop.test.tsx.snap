// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoTop /> should match snapshot 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  color: #FFFFFF;
  background-color: #0A1128;
}

<a
  aria-label="Go to top"
  class="c0"
  href="#"
  title="Go to top"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
    />
  </svg>
</a>
`;
